src/contexts/NetworkContext.js

```diff
@@ -0,0 +1,61 @@
+import React, { createContext, useState, useEffect, useContext } from 'react';
+import storage from '../services/storage';
+
+const NetworkConext = createContext();
+
+/**
+ * Custom hook to subscribe to network status.
+ *
+ * Value is retrieved from local storage initially and updated by service worker firing events
+ *
+ * Can be used by any functional component under the NetworkContextProvider in App.js
+ *
+ * @returns {boolean} true if connected to the network, false otherwise.
+ */
+export const useNetworkStatus = () => {
+  const context = useContext(NetworkConext);
+  if (context === undefined) {
+    throw new Error(`useNetworkStatus must be called within NetworkContextProvider`);
+  }
+
+  return context;
+};
+
+const NetworkContextProvider = ({ children }) => {
+  const [isOnline, setIsOnline] = useState('online');
+
+  useEffect(() => {
+    // Retrieve network status from local storage or default to online
```

Comment:
Why should it default to online?

Reply:
The short answer is that, if it defaults to offline, we risk showing a flash of the offline component tree to users who are actually online while we fetch the network status. Since ~95% of our users are exclusively using the app online, this is undesirable to say the least. And in the case that the app IS offline, the online component tree - which is essentially the default in users' mind - will be replaced with the offline version in a useful amount of time (usually imperceptibly quickly).